### ISSUE TITLE
i++ should be parsed as UpdateExpression/++ rather than UnaryExpression/PlusPlusToken

### ISSUE THREAD

**Issue by danvk** [eslint/typescript-eslint-parser#58]
I ran into this while debugging why this got flagged with eslint:

```ts
let i = 0;
function f() {
  i++;
}
f();
```

The error is `'i' is never reassigned. Use 'const' instead`.

The esprima parse for the `i++` looks like this:

```json
          {
            "expression": {
              "argument": {
                "name": "i",
                "type": "Identifier"
              },
              "operator": "++",
              "prefix": false,
              "type": "UpdateExpression"
            }
```

whereas the typescript-eslint-parser equivalent is:

```json
            "expression": {
              "argument": {
                "name": "i",
                "type": "Identifier"
              },
              "operator": "PlusPlusToken",
              "prefix": false,
              "type": "UnaryExpression"
            }
```

**Comment by JamesHenry** [comment]
Thanks for your digging here @danvk, you're exactly right. Fix submitted here: https://github.com/eslint/typescript-eslint-parser/pull/59
